### PR TITLE
OBPIH-6668 Responsiveness in very long shipments with split line appears and disappears

### PR DESCRIPTION
**Pull request by alannadolny**
### :sparkles: Description of Change

**Link to GitHub issue or Jira ticket:** [OBPIH-6668](https://pihemr.atlassian.net/browse/OBPIH-6668?atlOrigin=eyJpIjoiOTE3NjZjZDAxNTg2NGI1YWJjYTYxZTFiMmY5OGJhYmIiLCJwIjoiaiJ9)

**Description:**
The issue with disappearing new lines was caused by managing the state in the component and in the form state at the same time, so when we were fetching new lines, we overrode the already created new lines.

[OBPIH-6668]: https://pihemr.atlassian.net/browse/OBPIH-6668?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/js/components/stock-movement-wizard/combined-shipments/AddItemsPage.jsx [hunk 1]
```diff
@@ -350,29 +335,35 @@ class AddItemsPage extends Component {
...
+    append?.('lineItems', lineItemsData);
+
+    if (data.length < 10) {
       this.props.hideSpinner();
+      return;
+    }
+
+    this.loadMoreRows(append)({
+      startIndex: startIndex + this.props.pageSize,
+      fetchedLineItems: fetchedLineItems + lineItemsData.length,
```
Comment (kchelstowski) [inline src/js/components/stock-movement-wizard/combined-shipments/AddItemsPage.jsx:363]: at first thought I would think that `fetchedLineItems` actually store the items, not the size of the items, so how about renaming it to `fetchedItemsSize`/`fetchedLineItemsSize`/`fetchedLineItemsCount`?

**Review by kchelstowski — COMMENTED** [review]
as a side note: in some places it's a bit difficult to spot, what is actually changed and what are just the lint fixes, so I would not call the fix eslint on a whole file, or at least indicate in a PR what lines are the "real" changes" and which could be skipped.

File: src/js/components/stock-movement-wizard/combined-shipments/AddItemsPage.jsx (hunk 1 above)
Comment (alannadolny) [reply src/js/components/stock-movement-wizard/combined-shipments/AddItemsPage.jsx:363]: > changed and what are just the lint fixes, so I would not call the fix eslint on a whole file
btw, I think we should run the linter for the entire project some time ago.

Comment (alannadolny) [reply src/js/components/stock-movement-wizard/combined-shipments/AddItemsPage.jsx:363]:
``` 
fetchedLineItems: fetchedLineItems + lineItemsData.length,
```
I noticed that it's not used anymore, so I am deleting it :smile:

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**